browser/tor/test/onion_location_navigation_throttle_browsertest.cc

```diff
@@ -174,28 +174,50 @@ class OnionLocationNavigationThrottleBrowserTest : public InProcessBrowserTest {
-    tor::OnionLocationTabHelper* helper =
-        tor::OnionLocationTabHelper::FromWebContents(web_contents);
-    EXPECT_TRUE(helper->should_show_icon());
-    EXPECT_EQ(helper->onion_location(), GURL(kTestOnionURL));
-    CheckOnionLocationLabel(browser, GURL(kTestOnionURL), false);
-
-    GURL url2 = test_server()->GetURL("/no_onion");
-    ASSERT_TRUE(ui_test_utils::NavigateToURL(browser, url2));
-    web_contents = browser->tab_strip_model()->GetActiveWebContents();
-    helper = tor::OnionLocationTabHelper::FromWebContents(web_contents);
-    EXPECT_FALSE(helper->should_show_icon());
-    EXPECT_TRUE(helper->onion_location().is_empty());
-    auto* onion_location_view = GetOnionLocationView(browser);
-    ASSERT_TRUE(onion_location_view);
-    EXPECT_FALSE(onion_location_view->GetVisible());
-  }
+  auto* browser = browser();
+  GURL url1 = test_server()->GetURL("/onion");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser, url1));
+  content::WebContents* web_contents =
+      browser->tab_strip_model()->GetActiveWebContents();
+  tor::OnionLocationTabHelper* helper =
+      tor::OnionLocationTabHelper::FromWebContents(web_contents);
+  EXPECT_TRUE(helper->should_show_icon());
+  EXPECT_EQ(helper->onion_location(), GURL(kTestOnionURL));
+  CheckOnionLocationLabel(browser, GURL(kTestOnionURL), false);
+
+  GURL url2 = test_server()->GetURL("/no_onion");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser, url2));
+  web_contents = browser->tab_strip_model()->GetActiveWebContents();
+  helper = tor::OnionLocationTabHelper::FromWebContents(web_contents);
+  EXPECT_FALSE(helper->should_show_icon());
+  EXPECT_TRUE(helper->onion_location().is_empty());
+  auto* onion_location_view = GetOnionLocationView(browser);
+  ASSERT_TRUE(onion_location_view);
+  EXPECT_FALSE(onion_location_view->GetVisible());
+}
+
+IN_PROC_BROWSER_TEST_F(OnionLocationNavigationThrottleBrowserTest,
+                       OnionLocationHeaderTor) {
+  auto* browser = OpenTorWindow();
+  GURL url1 = test_server()->GetURL("/onion");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser, url1));
+  content::WebContents* web_contents =
+      browser->tab_strip_model()->GetActiveWebContents();
+  tor::OnionLocationTabHelper* helper =
+      tor::OnionLocationTabHelper::FromWebContents(web_contents);
+  EXPECT_TRUE(helper->should_show_icon());
+  EXPECT_EQ(helper->onion_location(), GURL(kTestOnionURL));
+  CheckOnionLocationLabel(browser, GURL(kTestOnionURL), false);
+
+  GURL url2 = test_server()->GetURL("/no_onion");
+  ASSERT_TRUE(ui_test_utils::NavigateToURL(browser, url2));
+  web_contents = browser->tab_strip_model()->GetActiveWebContents();
+  helper = tor::OnionLocationTabHelper::FromWebContents(web_contents);
+  EXPECT_FALSE(helper->should_show_icon());
+  EXPECT_TRUE(helper->onion_location().is_empty());
+  auto* onion_location_view = GetOnionLocationView(browser);
+  ASSERT_TRUE(onion_location_view);
+  EXPECT_FALSE(onion_location_view->GetVisible());
```

Comment:
Most of the codes in these two suites are duplicate and it is suitable for the use of `INSTANTIATE_TEST_SUITE_P` with `::testing::Bool()` as `IsTorWindow`

Reply:
Yeah, right exactly , I realized that when I started falling asleep last night.